PowerEditor/src/Notepad_plus.cpp

```diff
@@ -3328,6 +3328,19 @@ void Notepad_plus::checkModifiedDocument(bool bCheckOnlyCurrentBuffer)
 	MainFileManager->checkFilesystemChanges(bCheckOnlyCurrentBuffer);
 }
 
+void Notepad_plus::checkModifiedDocumentIfMonitoring()
+{
+	// This is an workaround to deal with Microsoft issue in ReadDirectoryChanges notification
+	// If command prompt is used to write file continuously (e.g. ping -t 8.8.8.8 > ping.log)
+	// Then ReadDirectoryChanges does not detect the change
+	// Fortunately, notification is sent if right click or double click happens on that file
+	// Let's leverage this as workaround to enhance npp file monitoring functionality.
+
+	Buffer* currBuf = MainFileManager->getCurrentBuffer();
+	if (currBuf && currBuf->isMonitoringOn())
+		::PathFileExists(currBuf->getFullPathName());

```

Comment:
Does `PathFileExists` imitate the behavior of double/right click to the file?

Reply:
> Does PathFileExists imitate the behavior of double/right click to the file?

No, it does not. It just checks for file existence. But internally, I might be gathering some info such as file size, last modified, last access or something similar. So when this API is called, that time on Windows10, file info is updated automatically. So it is just being utilized here.

Reply:
I understand that this function is part of Shell API, and might be doing something internally to flush the existing open file buffers (Explorer is very close to OS/file-system). However, I couldn't find, anywhere, that this function does flush the file buffers of open files. Hence, this solution is not provably correct.
Also, it might be working only on Windows 10, and may not exhibit the same behavior on other OS. 
It should also be noted that file monitoring and antivirus software, which do install file-system drivers, might also result in this behavior.

Reply:
> Hence, this solution is not provably correct.

This is a work around. 

> Also, it might be working only on Windows 10, and may not exhibit the same behavior on other OS.
This occurs only in win10 (as of now). There is no issue observed in other OS. Something has been broken on windows 10.

Reply:
I understand now. Then this function should be called when OS >= Windows 10 only. However, I don't see `winVer` enum having Win10 as one of the value.